core/vm/validate.go

```diff
@@ -0,0 +1,214 @@
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package vm
+
+import (
+	"errors"
+	"fmt"
+	"io"
+)
+
+// Below are all possible errors that can occur during validation of
+// EOF containers.
+var (
+	ErrUndefinedInstruction     = errors.New("undefined instruction")
+	ErrTruncatedImmediate       = errors.New("truncated immediate")
+	ErrInvalidSectionArgument   = errors.New("invalid section argument")
+	ErrInvalidCallArgument      = errors.New("callf into non-returning section")
+	ErrInvalidDataloadNArgument = errors.New("invalid dataloadN argument")
+	ErrInvalidJumpDest          = errors.New("invalid jump destination")
+	ErrInvalidBackwardJump      = errors.New("invalid backward jump")
+	//ErrConflictingStack              = errors.New("conflicting stack height")
+	//ErrInvalidBranchCount            = errors.New("invalid number of branches in jump table")
+	ErrInvalidOutputs                = errors.New("invalid number of outputs")
+	ErrInvalidMaxStackHeight         = errors.New("invalid max stack height")
+	ErrInvalidCodeTermination        = errors.New("invalid code termination")
+	ErrEOFCreateWithTruncatedSection = errors.New("eofcreate with truncated section")
+	ErrOrphanedSubcontainer          = errors.New("subcontainer not referenced at all")
+	ErrIncompatibleContainerKind     = errors.New("incompatible container kind")
+	ErrStopAndReturnContract         = errors.New("Stop/Return and Returncontract in the same code section")
+	ErrStopInInitCode                = errors.New("initcode contains a RETURN or STOP opcode")
+	ErrTruncatedTopLevelContainer    = errors.New("truncated top level container")
+	ErrUnreachableCode               = errors.New("unreachable code")
+	ErrInvalidNonReturningFlag       = errors.New("invalid non-returning flag, bad RETF")
```

Comment:
Not all of these methods are mapped in the eofparser error-to-number mapping function. Should they be?

Reply:
At some point maybe, for now we've given up on making eofparser match the error numbers